src/mame/drivers/mekd1.cpp

```diff
@@ -0,0 +1,462 @@
+		m_bit_rate_half_out = 0;
+		m_bit_rate_reset = 1;
+		return;
+	}
+
+	if (m_bit_rate_reset)
+	{
+		// Timer has just been taken out of reset.
+		m_bit_rate_reset = 0;
+		uint16_t delay = m_baud_rate->read();
+		// An adjustment is subtracted from the delay for the CPU
+		// instruction processing paths as the CPU polls the timer
+		// output, and this would vary with the CPU clock. In
+		// hardware, variable resisters needed to be tuned to achieve
+		// the correct timing which was not hard at the lower baud
+		// rates.
+		m_bit_rate_timer->adjust(attotime::from_usec(delay - 31));
+		m_bit_rate_half_timer->adjust(attotime::from_usec((delay / 2 ) - 31));
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::pia0_cb2_w)
+{
+	// This is a tape reader control line.
+}
+
+void mekd1_state::device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr)
+{
+	switch (id)
+	{
+	case TIMER_BIT_RATE:
+		m_bit_rate_out = 1;
+		break;
+	case TIMER_BIT_RATE_HALF:
+		m_bit_rate_half_out = 1;
+		break;
+	default:
+		throw emu_fatalerror("Unknown id in mekd1_state::device_timer");
+	}
+}
+
+
+WRITE_LINE_MEMBER(mekd1_state::write_f1_clock)
+{
+	if (m_acia_baud_rate->read() == 1) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f2_clock)
+{
+	if (m_acia_baud_rate->read() == 2) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f4_clock)
+{
+	if (m_acia_baud_rate->read() == 4) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f5_clock)
+{
+	if (m_acia_baud_rate->read() == 5) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f7_clock)
+{
+	if (m_acia_baud_rate->read() == 7) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f8_clock)
+{
+	if (m_acia_baud_rate->read() == 8) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f9_clock)
+{
+	if (m_acia_baud_rate->read() == 9) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f11_clock)
+{
+	if (m_acia_baud_rate->read() == 11) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+WRITE_LINE_MEMBER(mekd1_state::write_f13_clock)
+{
+	if (m_acia_baud_rate->read() == 13) {
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
+}
+
+
+void mekd1_state::machine_reset()
+{
+	m_pia0->reset();
+	m_pia1->reset();
+	m_brg->rsa_w(CLEAR_LINE);
+	m_brg->rsb_w(ASSERT_LINE);
+	m_bit_rate_timer->reset();
+	m_bit_rate_half_timer->reset();
+	m_bit_rate_out = 0;
+	m_bit_rate_half_out = 0;
+	m_bit_rate_select = 0;
+	m_bit_rate_reset = 1;
+}
+
+void mekd1_state::machine_start()
+{
+	// Allocate timers
+	m_bit_rate_timer = timer_alloc(TIMER_BIT_RATE);
+	m_bit_rate_half_timer = timer_alloc(TIMER_BIT_RATE_HALF);
+}
```

Comment:
Please register state members for save/load.  We want to get to a point where machines/devices can be expected to support save states by default.